### PR TITLE
Update config options without restarting

### PR DESCRIPTION
**Pull request by xconverge**
This is really nice and was easy!

It exposes some weirdness with how we do remapping but o well.. let me know what you think

### CODE REVIEW

File: src/configuration/configuration.ts [hunk 1]
```diff
@@ -149,28 +153,28 @@ class ConfigurationClass {
   /**
    * Size of a tab character.
    */
-  @overlapSetting({ codeName: "tabSize", default: 8})
+  @overlapSetting({ codeName: "tabSize", default: 8 })
```
Comment (johnfn) [inline src/configuration/configuration.ts]: Haha I see my weird spacing habits are starting to rub off on you.

**Comment by johnfn** [comment]
Overall feedback: Instead of updating them, could you just recreate them entirely - construct new ones? Ideally we only have one code path: addNewRemappers(), which gets called immediately and whenever a config update happens, rather than the 2 code paths we have right now.

File: src/configuration/configuration.ts (hunk 1 above)
Comment (xconverge) [inline src/configuration/configuration.ts]: cmd+a then = :)

**Comment by xconverge** [comment]
yep, makes sense

File: src/configuration/configuration.ts (hunk 1 above)
Comment (johnfn) [inline src/configuration/configuration.ts]: Whoa, I didn't know that it was the standard style... I thought I was doing something weird.

**Comment by rebornix** [comment]
LGTM, the only thing that I think we can change is guarding Vim from flooded configuration changes.

**Comment by johnfn** [comment]
Yeah this is awesome. It's super simple and an incredibly useful change. Imagine how many fewer questions we'll have to answer on Slack now 😉

**Pull request merged**